contracts/xasset/src/token.rs

```diff
@@ -1198,51 +1366,78 @@ impl Token {
                 accrued_interest: decorated_cdp.accrued_interest,
             },
         );
-    }
-
-    fn native(&self) -> token::Client {
-        token::Client::new(
-            env(),
-            &self
-                .xlm_sac
-                .get()
-                .expect("XLM Stellar Asset Contract should be initialized"),
-        )
-    }
 
-    // convenience functions for internal minting / transfering of the ft asset
-    fn mint_internal(&mut self, to: Address, amount: i128) {
-        let balance = self.balance(to.clone()) + amount;
-        self.balances.set_and_extend(to, &balance);
+        // Extend TTL
+        let ttl = env.storage().max_ttl();
+        env.storage()
+            .persistent()
+            .extend_ttl(&DataKey::CDP(lender.clone()), ttl, ttl);
     }
 
-    fn transfer_internal(&mut self, from: Address, to: Address, amount: i128) {
-        let from_balance = self.balance(from.clone()) - amount;
-        let to_balance = self.balance(to.clone()) + amount;
-        self.balances.set_and_extend(from, &from_balance);
-        self.balances.set_and_extend(to, &to_balance);
+    fn native(&self, env: &Env) -> TokenClient {
+        TokenClient::new(env, &Self::xlm_sac(env))
     }
 
-    fn burn_internal(&mut self, from: Address, amount: i128) {
-        let balance = self.balance(from.clone()) - amount;
-        self.balances.set_and_extend(from, &balance);
+    // convenience functions for internal minting / transfering of the ft asset
+    fn mint_internal(&self, env: &Env, to: Address, amount: i128) {
+        let balance = env
+            .storage()
+            .persistent()
+            .get(&DataKey::Balance(to.clone()))
+            .unwrap_or(0);
+        let new_balance = balance + amount;
+        env.storage()
+            .persistent()
+            .set(&DataKey::Balance(to.clone()), &new_balance);
+    }
+
+    fn transfer_internal(&self, env: &Env, from: Address, to: Address, amount: i128) {
```

Comment:
```suggestion
    fn transfer_internal(env: &Env, from: Address, to: Address, amount: i128) {
```

You should drop the `self`s everywhere since now they are static methods. Then just `Self::satic_method(env,...)`. The only reason we had self before was because that's how storage was accessed.